### PR TITLE
netdev_ieee802154/radios: refactor PAN ID reset to generic ieee802154 reset

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR refactors the PAN ID from the radio reset function to the generic `netdev_ieee802154_reset` and sets it via the `netdev::set` call. This allows for removal of the `dev->netdev` accesses in the radio specific `set_pan` calls.

### Testing procedure

Verify that initialization and reset of the PAN ID works for the affected radios

- [x] cc2538
- [x] socket_zep
- [x] at86rf2xx
- [x] cc2420
- [x] kw2xrf
- [x] mrf24j40

### Issues/PRs references

Chipping away at dependencies for #7736

### CODE REVIEW

File: drivers/netdev_ieee802154/netdev_ieee802154.c [hunk 1]
```diff
@@ -63,6 +63,9 @@ void netdev_ieee802154_reset(netdev_ieee802154_t *dev)
 #elif MODULE_GNRC
     dev->proto = GNRC_NETTYPE_UNDEF;
 #endif
+
+    static const uint16_t nid = IEEE802154_DEFAULT_PANID;
```
Comment (miri64) [inline drivers/netdev_ieee802154/netdev_ieee802154.c]: Why not set (and provide in the function below) `dev->pan` here?

**Comment by bergzand** [comment]
> One minor optimization proposal I'm not sure about: I think this can be removed:

Thanks, I already suspected that I was missing something in that driver.

File: cpu/cc2538/radio/cc2538_rf_netdev.c [hunk 2]
```diff
@@ -361,8 +361,6 @@ static int _init(netdev_t *netdev)
     netdev_ieee802154_reset(&dev->netdev);
 
     /* Initialise netdev_ieee802154_t struct */
-    netdev_ieee802154_set(&dev->netdev, NETOPT_NID,
-                          &pan, sizeof(pan));
```
Comment (miri64) [inline cpu/cc2538/radio/cc2538_rf_netdev.c]: The [line above](https://github.com/RIOT-OS/RIOT/pull/10384/files#diff-b8d088b7424905055347a5d1f7bd781cR356) isn't needed anymore ;-)

Comment (bergzand) [reply cpu/cc2538/radio/cc2538_rf_netdev.c]: Removed!

File: drivers/netdev_ieee802154/netdev_ieee802154.c (hunk 1 above)
Comment (bergzand) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: Changed (reduces flash size by 4 whole bytes on a samr21-xpro  :D )

Comment (miri64) [reply drivers/netdev_ieee802154/netdev_ieee802154.c]: Amazing!!1! :grin:

**Comment by miri64** [comment]
Also tested `at86rf2xx` with `gnrc_networking` and pinging (and checking `ifconfig`) on `iotlab-m3`

**Comment by miri64** [comment]
Murdock likes it (at least compilation-wise) so I guess we can put a virtual tick on that as well :-)

**Comment by miri64** [comment]
`cc2420` doesn't work (when using `example/default` it doesn't send packets that are recognized by an `at86rf2xx` device, multicast does not work and when a packet is sent from an `at86rf2xx` device it doesn't show the packet). But that's the case in master (not sure if there is already an issue for that). It sends however packets with the correct PAN ID and I confirmed with `ifconfig` in `examples/default` that the PAN ID is the one configured at compile time.

**Comment by miri64** [comment]
`kw2xrf` confirmed to work on `pba-d-01-kw2x`.

**Comment by miri64** [comment]
@PeterKietzmann can you have a look with the `cc2538` and the `mrf24j40` (and maybe confirm the bug in `cc2420` in master)?

**Comment by miri64** [comment]
@smlng do you have access to the missing devices?

**Comment by miri64** [comment]
(`cc2538` I'll test myself in the testbed ;-))

**Comment by smlng** [comment]
remote-revb (cc2538) works as expected

**Comment by miri64** [comment]
Tested on `firefly` (`cc2538`) with `example/default` (both by changing the PAN ID using `ifconfig` and by at compile time using `CFLAGS=-DIEEE802154_DEFAULT_PANID=0x6456`). `ifconfig` still shows the expected value and I was still able to exchange packages packets between the nodes if (and only if - as expecded) the the PAN ID was the same on both nodes. A sniffer running in parallel also showed the correct PAN ID for the exchanged packets.

**Comment by PeterKietzmann** [comment]
Tested with mrf24j40 + nucleo-l476rg and *gnrc_networking*:
- PANID after reset
- changing PAN ID
- verified by pinging a samr21-xpro with gnrc_networking and master

-> works as expected to I'll check the box above.

**Comment by bergzand** [comment]
Squashed, thanks for testing!

**Pull request merged**